### PR TITLE
[nemo-qml-plugin-systemsettings] Move the tests under /opt/tests/. Contributes to JB#56209

### PR DESCRIPTION
**Pull request by xiaobixu**

### CODE REVIEW

**Comment by pvuorela** [comment]
This does not yet move the actual tests, only the xml for what is executed on running the test case.

Need to adjust also tests/tests.pro for the target directory and the rpm .spec file for what files are included in the packages. 

And would be good if the template system from this file got removed, i.e. this @PACKAGENAME@ replacement, and this would be just a static file. The template does not provide nearly anything here, we are not constantly changing the package name.

**Comment by Thaodan** [comment]
> And would be good if the template system from this file got removed, i.e. this @packagename@ replacement, and this would be just a static file. The template does not provide nearly anything here, we are not constantly changing the package name.

The package name could be still different depending on how or who packages it. I don't see @packagename@ over complicating things here.

**Comment by pvuorela** [comment]
> The package name could be still different depending on how or who packages it. 

That's hard-coded in the tests.pro file so without further changes the packager cannot even affect it.

File: tests/tests.pro [hunk 1]
```diff
@@ -5,12 +5,12 @@ PACKAGENAME = nemo-qml-plugin-systemsettings
 QT += testlib qml dbus systeminfo
 QT -= gui
 
-system(sed -e s/@PACKAGENAME@/$${PACKAGENAME}/g tests.xml.template > tests.xml)
+# system(sed -e s/@PACKAGENAME@/$${PACKAGENAME}/g tests.xml.template > tests.xml)
```
Comment (pvuorela) [inline tests/tests.pro:8]: Remove the commented out line.

File: tests/tests.pro [hunk 2]
```diff
@@ -5,12 +5,10 @@ PACKAGENAME = nemo-qml-plugin-systemsettings
...
 
-system(sed -e s/@PACKAGENAME@/$${PACKAGENAME}/g tests.xml.template > tests.xml)
-
 TEMPLATE = app
 TARGET = ut_diskusage
 
-target.path = $$[QT_INSTALL_LIBS]/$${PACKAGENAME}-tests
+target.path = /opt/tests/$${PACKAGENAME}-tests
 
 xml.path = /usr/share/$${PACKAGENAME}-tests
```
Comment (pvuorela) [inline tests/tests.pro:13]: This is still the old directory while the .spec file looks for it in the spec. Change this also to the same path as above.

Comment (xiaobixu) [reply tests/tests.pro]: Fixed this as well :)

**Review by pvuorela — APPROVED** [review]
Looking good.

**Pull request merged**